js/searcher.js

```diff
@@ -232,35 +232,58 @@ async function findPlayer() {
-                    body: JSON.stringify(readyToFetch)
-                }).then(x => x.json()).then(res => {
-                    res.data.forEach(data => {
-                        if (data.imageUrl === avatarImageUrl) {
-                            foundUser.unshift(data.requestId);
-                            i = allTokens.length;
-                        }
-                    });
+        const maxParallelRequests = 20;
+        //checkUserBatch() returns true if the user is found, else returns false.
+        async function checkUserBatch(slice, attempts = 1) {
+            if (attempts > 3) return false; // Fail after 3 attempts
+            try {
+                const response = await fetch("https://thumbnails.roblox.com/v1/batch", {
+                    method: "POST",
+                    headers: {
+                        "Content-Type": "application/json",
+                        Accept: "application/json"
+                    },
+                    body: JSON.stringify(slice)
                 });
-
-                PROGRESS_BAR.style.width = `${Math.floor(i / onePercent) + 1}%`;
-                readyToFetch = [];
-                checked = 0;
+                const res = await response.json();
+                if (!res.data) throw new Error("No response data");
+                const targetUserData = res.data.find(data => data.imageUrl === avatarImageUrl);
+                if (targetUserData) {
+                    foundUser.unshift(targetUserData.requestId);
+                    return true;//User found
+                }
+                return false;//User not found in this slice
+            } catch (error) {
                 await sleep(50);
+                return await checkUserBatch(slice, attempts + 1);//Retry
+            }
+        }
+        //processSlices() returns true if any checkUserBatch() returned true otherwise returns false.
+        async function processSlices(slices) {
+            const results = await Promise.all(slices.map(slice => checkUserBatch(slice)));
+            return results.some(result => result === true);
+        }
+        for (let currentIndex = 0;
+            currentIndex < allTokens.length;
+            currentIndex += maxParallelRequests * 100) {
+            /*
+            slice = [allTokens[x], ...98x , allTokens[ (x+99) | (allTokens.length-1) ] ]
+            slices = [ slice, ...(maxParrallelRequests-2)x, slice ];
+            */
+            const slices = [];
+            for (let requestIndex = 0; requestIndex < maxParallelRequests; requestIndex++) {
+                const start = currentIndex + requestIndex * 100;
+                if (start >= allTokens.length) break;
+                slices.push(allTokens.slice(start, Math.min(start + 100, allTokens.length)));
             }
+            const userFound = await processSlices(slices);
+            if (userFound) return foundUser;
+            PROGRESS_BAR.style.width = `${Math.ceil(currentIndex / onePercent)}%`;
```

Comment:
The progress bar isn't updating accurately with the new changes. It jumps from 0% to 50% to 100% when processing 4000 tokens because it updates every 2000 tokens (maxParallelRequests * 100). Can you make it update more frequently for smoother progress tracking?

I'll merge this one, but can you fix it? If not, I will fix it later. Thanks.